paper/src/main/java/vg/civcraft/mc/civmodcore/commands/CommandManager.java

```diff
@@ -59,30 +77,22 @@ public final void init() {
 	 * This is called as part of {@link CommandManager#init()} and should be overridden by an extending class to
 	 * register all (or as many) commands at once.
 	 */
-    public void registerCommands() { }
+	public void registerCommands() {
+	}
 
-    /**
+
+	/**
 	 * This is called as part of {@link CommandManager#init()} and should be overridden by an extending class to
 	 * register all (or as many) completions at once, though make sure to call super.
 	 *
 	 * @param completions The completion manager is given. It is the same manager that can be reached via
 	 *                    {@link #getCommandCompletions()}.
 	 */
-    public void registerCompletions(@Nonnull final CommandCompletions<BukkitCommandCompletionContext> completions) {
+	public void registerCompletions(@Nonnull final CommandCompletions<BukkitCommandCompletionContext> completions) {
 		completions.registerCompletion("none", (context) -> Collections.emptyList());
-		completions.registerAsyncCompletion("allplayers", (context) ->
-				Arrays.stream(Bukkit.getOfflinePlayers())
-						.map(OfflinePlayer::getName)
-						.toList());
-		completions.registerAsyncCompletion("materials", (context) ->
-				Arrays.stream(Material.values())
-						.map(Enum::name)
-						.toList());
-		completions.registerAsyncCompletion("itemMaterials", (context) ->
-				Arrays.stream(Material.values())
-						.filter(ItemUtils::isValidItemMaterial)
-						.map(Enum::name)
-						.toList());
+		completions.registerAsyncCompletion("allplayers", (context) -> new ArrayList<>(autocompletePlayerNames));
```

Comment:
Remove the dependency on the concurrent skiplist which may change while being accessed by other code. This is a snapshotting basically and costs about nothing compared to anything else going on around it. Could be collections.immutablelist, but I haven't looked up the outer calling code, so if that wants to modify the list, it can, but not the original skiplist. Which should be preserved as far as I can see.

Reply:
Comment added to clarify the intent here.